drivers/overlay/overlay.go

```diff
@@ -157,32 +158,71 @@ func Init(home string, options graphdriver.Options) (graphdriver.Driver, error)
 	if err := idtools.MkdirAllAs(path.Join(home, linkDir), 0700, rootUID, rootGID); err != nil && !os.IsExist(err) {
 		return nil, err
 	}
+	runhome := filepath.Join(options.RunRoot, filepath.Base(home))
+	if err := idtools.MkdirAllAs(runhome, 0700, rootUID, rootGID); err != nil && !os.IsExist(err) {
+		return nil, err
+	}
 
 	var usingMetacopy bool
 	var supportsDType bool
 	if opts.mountProgram != "" {
 		supportsDType = true
 	} else {
-		supportsDType, err = supportsOverlay(home, fsMagic, rootUID, rootGID)
-		if err != nil {
-			os.Remove(filepath.Join(home, linkDir))
-			os.Remove(home)
-			patherr, ok := err.(*os.PathError)
-			if ok && patherr.Err == syscall.ENOSPC {
-				return nil, err
+		feature := "overlay"
+		overlayCacheResult, overlayCacheText, err := cachedFeatureCheck(runhome, feature)
+		if err == nil {
+			if overlayCacheResult {
+				logrus.Debugf("cached value indicated that overlay is supported")
+			} else {
+				logrus.Debugf("cached value indicated that overlay is not supported")
+			}
+			supportsDType = overlayCacheResult
+			if !supportsDType {
+				return nil, errors.New(overlayCacheText)
+			}
+		} else {
+			supportsDType, err = supportsOverlay(home, fsMagic, rootUID, rootGID)
+			if err != nil {
+				os.Remove(filepath.Join(home, linkDir))
+				os.Remove(home)
+				patherr, ok := err.(*os.PathError)
+				if ok && patherr.Err == syscall.ENOSPC {
+					return nil, err
+				}
+				if err2 := cachedFeatureRecord(runhome, feature, false, fmt.Sprintf("kernel does not support overlay fs: %v", err)); err2 != nil {
```

Comment:
This should be set to a variable so that every time you run command it outputs same error.

fmt.Sprintf("kernel does not support overlay fs: %v", err))

Reply:
This error should be fixed now.  The error reporting when the driver determines that the kernel or underlying filesystem won't work is messier, but I'd like to tackle that later.